common/repo/GitRepo.java

```diff
@@ -248,107 +225,73 @@ public void tagHead(String name) {
-      }
-      AddCommand addCommand = jGit.add();
-      jGit.merge().include(ref).call();
-      for (String conflictingFile : jGit.status().call().getConflicting()) {
-        successfulMerge = false;
-        addCommand = addCommand.addFilepattern(conflictingFile);
-      }
-      if (!successfulMerge) {
-        addCommand.call();
-        jGit.commit().call();
-      }
-      jGit.reset().setRef(current.getObjectId().toObjectId().name()).call();
-      return successfulMerge;
-    } catch (GitAPIException | IOException e) {
-      throw new RuntimeException(e);
-    }
+    runCommand("pull");
   }
 
+  @Override
   public boolean merge(String branch) {
     return merge(branch, false);
   }
 
   @Override
   public boolean mergeTheirs(String branch) {
-    try {
-      Ref current = jGitRepo.exactRef(jGitRepo.getFullBranch());
-      Ref ref = jGitRepo.exactRef("refs/heads/" + branch);
-      jGit.merge().setStrategy(MergeStrategy.THEIRS).include(ref).call();
-      jGit.reset().setRef(current.getObjectId().toObjectId().name()).call();
-      AddCommand add = jGit.add();
-      for (String file : jGit.status().call().getUntracked()) {
-        add.addFilepattern(file);
-      }
-      try {
-        add.call();
-        jGit.commit().setAll(true).setMessage("Updated changes").call();
-        return true;
-      } catch (NoFilepatternException e) {
-        // Nothing to commit
-        return true;
-      }
-    } catch (GitAPIException | IOException e) {
-      throw new RuntimeException(e);
+    switchToMasterBranch();
+    CommandResult commandResult = runCommand("merge " + branch + " -s recursive -Xtheirs");
+    String err = commandResult.stderr;
+    return err.length() == 0;
+  }
+
+  public boolean merge(String branch, boolean remote) {
+    switchToMasterBranch();
+    CommandResult commandResult;
+    if (remote) {
+      runCommand("fetch origin " + branch);
+      commandResult =
+          runCommand(
+              "merge origin/"
+                  + branch
+                  + " -m Merge_remote-tracking_branch_\'origin/"
+                  + branch
+                  + "\'");
+    } else {
+      commandResult = runCommand("merge " + branch);
     }
+    String err = commandResult.stderr;
+    return err.length() == 0;
   }
 
+  @Override
   public boolean isMerged(String branch) {
     throw new UnsupportedOperationException("Not implemented");
   }
 
+  @Override
   public void reset(String ref) {
-    try {
-      // MIXED type means that HEAD pointer would be
-      // reset to `ref` and all changes introduced after it
-      // would be marked as unstaged but saved in working tree
-      jGit.reset().setMode(ResetType.MIXED).setRef(ref).call();
-    } catch (GitAPIException e) {
-      throw new RuntimeException(e);
-    }
+    // HEAD pointer would be
+    // reset to `ref` and all changes introduced after it
+    // would be marked as unstaged but saved in working tree
+    runCommand("reset " + ref);
```

Comment:
We don't need it mixed anymore?
@vmax WDYT?

Reply:
@oferb `mixed` is the default mode (*). we could state it explicitly if you so wish but it won't change the logic

>(*) If you specify the --mixed option, reset will stop at this point. This is also the default, so if you specify no option at all (just git reset HEAD~ in this case), this is where the command will stop.
>
>from https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified

Reply:
Ok, great :)
No need to state it then.

Reply:
Done.